Добавление документации

Tarasusrus opened the pull request:
- добавлены доки

Tarasusrus committed:
Добавлен OpenAPI спецификация интерфейса для социальной сети путешественников

Tarasusrus committed:
Обновлена OpenAPI спецификация: добавлены новые эндпоинты для управления публикациями, комментариями и реакциями; обновлены схемы данных; применены изменения структуры документа для улучшения читаемости и соответствия новой функциональности.

Tarasusrus committed:
Обновлен файл README.md: исправлена ошибка в расчете трафика для POST запросов (Посты); добавлено описание модели хранения данных, а также расчет количества дисков для хранения и обработки данных за год. Добавлена новая диаграмма для визуализации модели хранения.

BalunCourses commented on README.md:
````diff
@@ -109,3 +109,51 @@
 
 ```
+
+### Модель хранения данных
````
Нужно делать в отдельном PR

BalunCourses commented on docs/doc.yaml:
```diff
@@ -0,0 +1,465 @@
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PublicationCreate'
+      responses:
```
А 500 здесь и ниже?

BalunCourses commented on docs/doc.yaml:
```diff
@@ -0,0 +1,465 @@
+        '401': { description: Unauthorized }
+        '404': { description: Comment not found }
+
+  /reactions:
```
А как получить? Не будем же мы тысячи реакций тянуть вместе с постом?

BalunCourses commented on docs/doc.yaml:
```diff
@@ -0,0 +1,465 @@
+        '401': { description: Unauthorized }
+
+
+
```
А медиа как грузить? И где подписки?

Tarasusrus committed:
Добавлены ручки для реакций, загрузки медиа и подписок (follow/unfollow, списки). Добавлен общий schema Error, в ключевые ручки добавлен ответ 500.

Tarasusrus committed:
Исключил решение 4дз

BalunCourses commented on docs/doc.yaml:
```diff
@@ -314,6 +403,166 @@ paths:
+          content:
+            application/json:
+              schema: { $ref: '#/components/schemas/Error' }
+  /reactions/summary:
```
А это можно сразу с постом отдавать, если эта ручка кол-во возвращает только

Tarasusrus committed:
**Добавлена поддержка реакции в API комментариев**
- В схему `Comment` добавлено поле `reactions` для отображения разбивки реакций.
- В схему `CommentCreate` также добавлено поле `reactions`.
- Создана новая схема `ReactionBreakdown` для описания структуры реакций.
- Актуализирована документация (doc.yaml) в соответствии с изменениями.

BalunCourses commented on README.md:
````diff
@@ -109,3 +109,51 @@
+![модель](/diagrams/ССДП%20домашка%203.svg)
+#### Расчет сколько потребуется дисков для хранения и обработки всех данных приложения на 1 год
+```
+Replicatio Factor = 3 
````
Еще не нужно было делать

BalunCourses commented on README.md:
```diff
@@ -109,3 +109,51 @@
+    Disks_for_capacity = 0 (лента не хранит)
+    Disks_for_throughput = 18 MB/S / 100 MB/S = 1 
+    Disks_for_iops = 579 / 100 = 6
+    rf = 18 дисков 
```
Каких именно дисков здесь и ниже?

BalunCourses commented on README.md:
````diff
@@ -109,3 +109,51 @@
+    Disks_for_throughput = 2KB/s / 100 MB/s = 1 
+    Disks_for_iops = 52/ 100 = 1
+    rf = 3 диска     
+```
````
Модель данных нужно кодом описать
